src/main/java/com/getindata/connectors/http/internal/table/lookup/HttpLookupTableSource.java

```diff
@@ -94,21 +100,41 @@ public LookupRuntimeProvider getLookupRuntimeProvider(LookupContext lookupContex
         PollingClientFactory<RowData> pollingClientFactory =
             createPollingClientFactory(lookupQueryCreator, lookupConfig);
 
-        HttpTableLookupFunction dataLookupFunction =
-            new HttpTableLookupFunction(
-                pollingClientFactory,
-                responseSchemaDecoder,
-                lookupRow,
-                lookupConfig
-            );
+        // In line with the JDBC implementation and current requirements, we are only
+        // supporting Partial Caching for synchronous operations.
+        return getLookupRuntimeProvider(lookupRow, responseSchemaDecoder, pollingClientFactory);
+    }
 
+    protected LookupRuntimeProvider getLookupRuntimeProvider(LookupRow lookupRow,
+        DeserializationSchema<RowData> responseSchemaDecoder,
+        PollingClientFactory<RowData> pollingClientFactory) {
         if (lookupConfig.isUseAsync()) {
+            HttpTableLookupFunction dataLookupFunction =
+                    new HttpTableLookupFunction(
+                            pollingClientFactory,
+                            responseSchemaDecoder,
+                            lookupRow,
+                            lookupConfig
+                    );
             log.info("Using Async version of HttpLookupTable.");
             return AsyncTableFunctionProvider.of(
                 new AsyncHttpTableLookupFunction(dataLookupFunction));
         } else {
-            log.info("Using blocking version of HttpLookupTable.");
-            return TableFunctionProvider.of(dataLookupFunction);
+            CachingHttpTableLookupFunction dataLookupFunction =
```

Comment:
In theory HttpTableLookupFunction could be used when cache is configured. Original function should only be instrumented with caching layer. What is the purpose of introducing new CachingHttpTableLookupFunction?

Reply:
@MarekMaj . Thanks for your review. The problem was that the original HttpTableLookupFunction was not a LookupFunction,it is defined as `public class HttpTableLookupFunction extends TableFunction<RowData> `) , so could not be used in Flink's caching frame.  CachingHttpTableLookupFunction extends LookupFunction so can be used in the caching framework. Have I understood your concern here?

Reply:
@davidradl yes, thank you for pointing this out. I think in the future we could unify those implementations